Add support for reneg-sec option

oc243 opened the pull request.

oc243 commented:
@luxflux any chance of a review of this when you get some time pls?

luxflux commented on manifests/client.pp:
```diff
@@ -214,6 +218,7 @@
   $shared_ca            = undef,
   $custom_options       = {},
   $expire               = undef,
+  $reneg_sec            = '',
```
This does not match the documentation and IMHO a default value of `undef` is more consistent than an empty string. Could you change this?

luxflux commented on manifests/server.pp:
```diff
@@ -491,6 +495,7 @@
   $tls_auth                  = false,
   $tls_server                = false,
   $tls_client                = false,
+  $reneg_sec                 = '',
```
Same here.

luxflux commented on spec/defines/openvpn_client_spec.rb:
```diff
@@ -114,6 +115,7 @@
     it { should contain_file('/etc/openvpn/test_server/download-configs/test_client/test_client.conf').with_content(/^verb\s+1$/)}
     it { should contain_file('/etc/openvpn/test_server/download-configs/test_client/test_client.conf').with_content(/^mute\s+10$/)}
     it { should contain_file('/etc/openvpn/test_server/download-configs/test_client/test_client.conf').with_content(/^auth-retry\s+interact$/)}
+    it { should contain_file('/etc/openvpn/test_server/download-configs/test_client/test_client.conf').with_content(/^reneg-sec\s+3600$/)}
```
Please also add the reverse test (it should not be there by default).

luxflux commented on spec/defines/openvpn_server_spec.rb:
```diff
@@ -211,6 +212,7 @@
 
     it { should contain_file('/etc/openvpn/test_server.conf').with_content(%r{^fragment 1412$}) }
     it { should contain_file('/etc/openvpn/test_server.conf').with_content(%r{^port-share 127.0.0.1 8443$}) }
+    it { should contain_file('/etc/openvpn/test_server.conf').with_content(/^reneg-sec\s+3600$/) }
```
Please also add the reverse test (it should not be there by default).

oc243 commented:
Hi @luxflux, thanks for the review. I think that the changes that I made to this branch on 13 Feb have addressed your points. Please may you take another look and see if they look good to you?

Thanks!

actatux commented on manifests/client.pp:
```diff
@@ -136,6 +136,10 @@
 #   Default: undef
 #
+# [*reneg_sec*]
+#   Integer. Value for how often to regenotiate keys.
```
Typo: renegotiate

actatux commented on manifests/server.pp:
```diff
@@ -305,6 +305,10 @@
 #   Default: false
 #
+# [*reneg_sec*]
+#   Integer. Value for how often to regenotiate keys.
```
Typo: renegotiate

bastelfreak commented:
Hi @oc243, can you please rebase this PR?

bastelfreak commented:
Hi. I'm going to close this PR due to inactivity. Please reopen and rebase it if you're still interested in it.
